src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1515,6 +1515,41 @@ internal static bool TryGetSpecialTypeMember<TSymbol>(CSharpCompilation compilat
             return true;
         }
 
+#nullable enable
+
+        /// <summary>
+        /// This is a layer on top of the Compilation version that generates a diagnostic if the well known
+        /// member isn't found.
+        /// </summary>
+        internal Symbol? GetWellKnownTypeMember(WellKnownMember member, BindingDiagnosticBag diagnostics, SyntaxNode syntax)
+        {
+            return TryGetWellKnownTypeMember<Symbol>(this.Compilation, member, syntax, diagnostics, out var memberSymbol)
+                ? memberSymbol
+                : null;
+        }
+
+        internal static bool TryGetWellKnownTypeMember<TSymbol>(CSharpCompilation compilation, WellKnownMember wellKnownMember, SyntaxNode syntax, BindingDiagnosticBag diagnostics, [MaybeNullWhen(false)] out TSymbol symbol)
+            where TSymbol : Symbol
+        {
+            symbol = (TSymbol?)compilation.GetWellKnownTypeMember(wellKnownMember);
+            if (symbol is null)
+            {
+                MemberDescriptor descriptor = WellKnownMembers.GetDescriptor(wellKnownMember);
+                diagnostics.Add(ErrorCode.ERR_MissingPredefinedMember, syntax.Location, descriptor.DeclaringTypeMetadataName, descriptor.Name);
+                return false;
+            }
+
+            var useSiteInfo = GetUseSiteInfoForWellKnownMemberOrContainingType(symbol);
+            if (useSiteInfo.DiagnosticInfo != null)
+            {
+                diagnostics.ReportUseSiteDiagnostic(useSiteInfo.DiagnosticInfo, new SourceLocation(syntax));
+            }
+
```

Comment:
@AlekseyTs In `TryGetSpecialTypeMember` you wrote:

```csharp
            // No need to track assemblies used by special members or types. They are coming from core library, which 
            // doesn't have any dependencies.
            return true;
```

presumably we then do need tracking for `TryGetWellKnownTypeMember`. How do I add this tracking?

Thanks